ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -402,10 +402,23 @@ private void checkAndMaybeComplete() {
         routerMetrics.operationFailureWithUnsetExceptionCount.inc();
       }
       if (e != null) {
+        // best effort to update metrics if result is available
+        boolean encrypted = operationResult != null && operationResult.getBlobResult != null
+            && operationResult.getBlobResult.getBlobInfo() != null && operationResult.getBlobResult.getBlobInfo()
+            .getBlobProperties()
+            .isEncrypted();
         operationResult = null;
-        routerMetrics.onGetBlobError(e, options);
+        routerMetrics.onGetBlobError(e, options, encrypted);
+      }
+      long operationLatencyMs = time.milliseconds() - submissionTimeMs;
+      routerMetrics.getBlobInfoOperationLatencyMs.update(operationLatencyMs);
```

Comment:
if you update the main one anyway, wouldn't it be against our goal of trying to separate the encrypted from the not encrypted metrics?

Reply:
Same comment in all other ops - want to understand what the intent here since in the frontend you separate them but in the router it looks like one metric contains both and another has just encrypted.

Reply:
yes. I did start the patch with that in mind. But for gets, we may not know if the blob is encrypted or not until we get the response. So, I chose this path. regular metrics will include both non-encrypted and encrypted blobs. encryption metrics will be emitted in addition for encrypted blobs.

Reply:
oh yeah. I missed that part. Once we have a consensus, will fix the diff between frontend metrics and router metrics.